airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -58,11 +71,28 @@ public OAuthConsentRead getSourceOAuthConsent(final SourceOauthConsentRequest so
     final Map<String, Object> metadata = generateSourceMetadata(sourceDefinitionIdRequestBody.getSourceDefinitionId());
     final OAuthConsentRead result;
     if (OAuthConfigSupplier.hasOAuthConfigSpecification(spec)) {
+      JsonNode oAuthInputConfigurationForConsent;
+
+      if (sourceDefinitionIdRequestBody.getSourceId() == null) {
+        oAuthInputConfigurationForConsent = sourceDefinitionIdRequestBody.getoAuthInputConfiguration();
+      } else {
+        final SourceConnection hydratedSourceConnection =
+            secretsRepositoryReader.getSourceConnectionWithSecrets(sourceDefinitionIdRequestBody.getSourceId());
+
+        List<String> fieldsToGet =
+            buildJsonPathFromOAuthFlowInitParameters(spec.getAuthSpecification().getOauth2Specification().getOauthFlowInitParameters());
```

Comment:
Why are we pulling these fields from the `spec.getAuthSpecification()`? Isn't that the old/legacy way of configuring OAuth?

The if-statement above which does `OAuthConfigSupplier.hasOAuthConfigSpecification(spec)` is checking if `advancedAuth` is set on the spec, which is the new way of configuring OAuth. So it seems like we shouldn't be accessing the legacy auth config fields here right?

Reply:
Done